src/primitives/transaction.cpp

```diff
@@ -145,9 +145,10 @@ std::string CTxOut::ToString() const
 {
     return strprintf("CTxOut(nValue=%d.%08d, scriptPubKey=%s)", nValue / COIN, nValue % COIN, scriptPubKey.ToString().substr(0,30));
 }
-
-CMutableTransaction::CMutableTransaction() : nVersion(CTransaction::MIN_CURRENT_VERSION), nLockTime(0) {}
-CMutableTransaction::CMutableTransaction(const CTransaction& tx) : nVersion(tx.nVersion), vin(tx.vin), vout(tx.vout), nLockTime(tx.nLockTime),
+// FIXME: we need a default value for branch id and expiry height, or leave it as 0?
```

Comment:
Default branch ID should match the default version, so if we default to v1 then use 0. I'm not quite sure what we do about the transition; probably just leave it defaulting to the Sprout version until after the upgrade is live (so manually setting Overwinter versions everywhere), and then we can change it once Overwinter has activated (and clean up the usage everywhere).

Default expiry height should be somewhere in #2874.

Reply:
Agreed.  I think for now we leave the defaults to v1.